### PR TITLE
[Fix] pUSD comments

### PR DESCRIPTION
**Pull request by ungaro**
## What's new in this PR?

Resolved comments for pUSD

### CODE REVIEW

**Comment by ungaro** [comment]
need to fix redeem function and test, other than that finished. @eyqs

**Comment by ungaro** [comment]
can i get your review, changes are finished. @eyqs

File: nest/src/token/pUSD.sol [hunk 1]
```diff
@@ -259,12 +242,12 @@ contract pUSD is
         address receiver,
         address controller,
         uint256 minimumMint
-    ) public virtual nonReentrant returns (uint256 shares) {
+    ) public virtual returns (uint256 shares) {
```
Comment (eyqs) [inline nest/src/token/pUSD.sol:245]: Why did you remove nonReentrant? (Might be because of my merged commit)

File: nest/src/token/pUSD.sol [hunk 2]
```diff
@@ -295,51 +278,70 @@ contract pUSD is
...
+     * @param price Price in terms of underlying asset (e.g., USDC) per share
+     * @param deadline Deadline for the atomic request
      */
-    function redeem(
+    function requestRedeem(
         uint256 shares,
         address receiver,
         address controller,
         uint256 price,
         uint64 deadline
```
Comment (eyqs) [inline nest/src/token/pUSD.sol:293]: Doesn't match the `IComponentToken` interface - there shouldn't be a price or a deadline

File: nest/src/token/pUSD.sol [hunk 3]
```diff
@@ -295,51 +278,70 @@ contract pUSD is
...
+     * @param shares Amount of shares to redeem
+     * @param receiver Address that will receive the assets
+     * @param controller Address that controls the redemption
+     */
+    function redeem(
+        uint256 shares,
+        address receiver,
+        address controller
+    ) public virtual override(ComponentToken) returns (uint256 assets) {
+        // Check claimableRedeemRequest, Transfer assets to receiver, Clean up request state.
```
Comment (eyqs) [inline nest/src/token/pUSD.sol]: This isn't fully implemented yet?

File: nest/src/interfaces/IBoringVault.sol [hunk 4]
```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.25;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { IERC20Metadata } from "@openzeppelin/contracts/token/ERC20/extensions/IERC20Metadata.sol";
+
+interface IVault is IERC20, IERC20Metadata {
```
Comment (eyqs) [inline nest/src/interfaces/IBoringVault.sol:7]: `interface IBoringVault`

File: nest/script/UpgradepUSD.s.sol
Comment (eyqs) [inline nest/script/UpgradepUSD.s.sol:1]: Replace `vm.assume` with `vm.skip` modifier

File: nest/src/ComponentToken.sol
Comment (eyqs) [inline nest/src/ComponentToken.sol:1]: Remove `_BASE` from definition of `AggregateToken.sol` because it can use `_BASE` from `ComponentToken.sol`

File: nest/src/token/pUSD.sol [hunk 5]
```diff
@@ -4,49 +4,33 @@ pragma solidity ^0.8.25;
...
-
 import { ERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
 import { ERC4626Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/extensions/ERC4626Upgradeable.sol";
-import { ERC20 } from "@solmate/tokens/ERC20.sol";
+import { IERC20 } from "@openzeppelin/contracts/interfaces/IERC20.sol";
+import { IERC20Metadata } from "@openzeppelin/contracts/interfaces/IERC20Metadata.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 
-import { IComponentToken } from "../interfaces/IComponentToken.sol";
+import { ERC20 } from "@solmate/tokens/ERC20.sol";
```
Comment (eyqs) [inline nest/src/token/pUSD.sol:13]: Is there any way to use `IERC20` instead of Solmate's `ERC20`?

File: nest/src/token/pUSD.sol (hunk 1 above)
Comment (eyqs) [reply nest/src/token/pUSD.sol:245]: Because of the comment above - No need to initialize ReentrancyGuard, as it's inherited from ComponentToken.

File: nest/src/token/pUSD.sol [hunk 6]
```diff
@@ -295,51 +278,70 @@ contract pUSD is
...
             inSolve: false
         });
 
-        IERC20(address(this)).safeIncreaseAllowance(address(queue), shares);
+        IAtomicQueue queue = _getpUSDStorage().boringVault.atomicQueue;
+        queue.updateAtomicRequest(IERC20(address(this)), IERC20(asset()), request);
 
-        // Update atomic request
-        queue.updateAtomicRequest(ERC20(address(this)), ERC20(asset()), request);
+        return 0; // ComponentToken's standard REQUEST_ID
```
Comment (eyqs) [inline nest/src/token/pUSD.sol:319]: `return REQUEST_ID;`

**Review by eyqs — APPROVED** [review]
LGTM! Only change needed for tonight before merging is making `requestRedeem` match the signature in `IComponentToken` and figuring out how to call `notifyRedeem` when the request is processed by the `AtomicQueue`.

File: nest/src/token/pUSD.sol (hunk 3 above)
Comment (eyqs) [reply nest/src/token/pUSD.sol]: All the logic is in the inherited code

File: nest/src/token/pUSD.sol (hunk 2 above)
Comment (ungaro) [reply nest/src/token/pUSD.sol:293]: but boringvault needs price and deadline for AtomicRequest in requestRedeem function? let me think about it.

File: nest/src/token/pUSD.sol (hunk 6 above)
Comment (ungaro) [reply nest/src/token/pUSD.sol:319]: changed

File: nest/src/interfaces/IBoringVault.sol (hunk 4 above)
Comment (ungaro) [reply nest/src/interfaces/IBoringVault.sol:7]: changed
